RecoLocalMuon/GEMSegment/plugins/GEMSegAlgo.cc

```diff
@@ -0,0 +1,479 @@
+    auto gemid = (*rh)->gemId();
+    // auto rhr = gemGeom->etaPartition(gemid);
+    auto rhLP = (*rh)->localPosition();
+    // auto rhGP = rhr->toGlobal(rhLP);
+    edm::LogVerbatim("GEMSegAlgo") << "[RecHit :: Loc x = "<<std::showpos<<std::setw(9)<<rhLP.x()<<" Loc y = "<<std::showpos<<std::setw(9)<<rhLP.y()
+				     <<" BX = "<<std::showpos<<(*rh)->BunchX()<<" -- "<<gemid.rawId()<<" = "<<gemid<<" ]";
+  }
+  #endif
+
+
+
+  // The actual fit on all hits of the vector of the selected Tracking RecHits:
+  sfit_ = std::unique_ptr<MuonSegFit>(new MuonSegFit(muonRecHits));
+  bool goodfit = sfit_->fit();
+  edm::LogVerbatim("GEMSegAlgo") << "[GEMSegAlgo::buildSegments] GEMSegment fit done :: fit is good = "<<goodfit;
+
+  // quit function if fit was not OK
+  if(!goodfit) return gemsegs;
+
+  // obtain all information necessary to make the segment:
+  LocalPoint protoIntercept      = sfit_->intercept();
+  LocalVector protoDirection     = sfit_->localdir();
+  AlgebraicSymMatrix protoErrors = sfit_->covarianceMatrix(); 
+  double protoChi2               = sfit_->chi2();
+
+  // Calculate the bunch crossing of the GEM Segment
+  float bx = 0.0;
+  for (auto rh=rechits.begin(); rh!=rechits.end(); ++rh){
+    bx += (*rh)->BunchX();
+  }
+  if(rechits.size() != 0) bx=bx*1.0/(rechits.size());
+
+  // Calculate the central value and uncertainty of the segment time
+  // if we want to study impact of 2-3ns time resolution on GEM Segment 
+  // (if there will be TDCs in readout and not just BX determination)
+  // then implement tof() method for rechits and use this here
+  /*`
+  float averageTime=0.;
+  for (auto rh=rechits.begin(); rh!=rechits.end(); ++rh){
+    GEMEtaPartition thePartition = theEnsemble.second.find((*rh)->gemId));
+    GlobalPoint pos = (thePartition->toGlobal((*rh)->localPosition());
+    float tof = pos.mag() * 0.01 / 0.2997925 + 25.0*(*rh)->BunchX(); 
+    averageTime += pos;                                          
+  }
+  if(rechits.size() != 0) averageTime=averageTime/(rechits.size());
+  float timeUncrt=0.;
+  for (auto rh=rechits.begin(); rh!=rechits.end(); ++rh){
+    GEMEtaPartition thePartition = theEnsemble.second.find((*rh)->gemId));
+    GlobalPoint pos = (thePartition->toGlobal((*rh)->localPosition());
+    float tof = pos.mag() * 0.01 / 0.2997925 + 25.0*(*rh)->BunchX();
+    timeUncrt += pow(tof-averageTime,2);
+  }
+  if(rechits.size() != 0) timeUncrt=timeUncrt/(rechits.size());
+  timeUncrt = sqrt(timeUncrt);
+  */
+
+  // save all information inside GEMCSCSegment
+  edm::LogVerbatim("GEMSegAlgo") << "[GEMSegAlgo::buildSegments] will now wrap fit info in GEMSegment dataformat";
+  // GEMSegment tmp(proto_segment, protoIntercept, protoDirection, protoErrors, protoChi2);
+  GEMSegment tmp(proto_segment, protoIntercept, protoDirection, protoErrors, protoChi2, bx);
+  // GEMSegment tmp(proto_segment, protoIntercept, protoDirection, protoErrors, protoChi2, averageTime, timeUncrt);
```

Comment:
I see how L470 may be needed. 
Why do we need L468 commented out alt construction?